json-schema/json-schema/src/main/scala/endpoints4s/Validated.scala

```diff
@@ -110,4 +110,18 @@ object Validated {
       case Failure(error) => Invalid(error.getMessage)
     }
 
+  /** Turns a `Seq[Validated[A]]`into a `Validated[Seq[A]]`.
+    * If all values in the sequence are `Valid`, the result will be `Valid`
+    * If there is at least one `Invalid`, the result will be an `Invalid` containing the error messages of all `Invalid`s
+    * If the `Seq` is empty, will return a `Valid[Seq.empty]`
+    */
+  def fromSeq[A](seq: Seq[Validated[A]]): Validated[Seq[A]] = {
+    seq.foldLeft(Valid(Seq.empty[A]): Validated[Seq[A]]) {
+      case (folded: Valid[Seq[A]], Valid(nextValid)) => folded.map(_ :+ nextValid)
+      case (_: Valid[Seq[A]], invalid: Invalid) => invalid
+      case (folded: Invalid, _: Valid[A]) => folded
+      case (Invalid(foldedErrors), Invalid(nextErrors)) => Invalid(foldedErrors ++ nextErrors)
```

Comment:
I think it would be more readable to use `zip` and `map` instead of re-implementing the behavior of `zip` here, what do you think?